geode-core/src/main/java/org/apache/geode/internal/security/SecurityServiceFactory.java

```diff
@@ -51,6 +53,11 @@ public static SecurityService create(Properties securityProps, CacheConfig cache
 
   public static SecurityService create(Properties securityProps,
       SecurityManager preferredSecurityManager, PostProcessor preferredPostProcessor) {
+    String securityServiceClassName = System.getProperty(SECURITY_SERVICE_SYSTEM_PROPERTY);
+    if (securityServiceClassName != null) {
```

Comment:
If this is truly for testing purposes only please isolate this code in its own method with some indicator in the name that it is an intended testing only smell.
```java
final SecurityManager testOnlySecurityManager = whateverThisMethodIsDoingForTesting();
if (null != testOnlySecurityManager) {
  return testOnlySecurityManager;
}
```
Though I would seriously question why we need something like this in production code? Perhaps there is a different way to write the tests that doesn't require special test only code in production. Could a custom security manager delegate be used that delegates to this testing only instance when necessary?

Reply:
We will need to simulate a cache closed scenario while calling the "authorized" method, this can't be done in the `SecurityManager`. The only way to reproduce this in a predicable way is to plug in a test security service object. I have tried testing it the other way, none of them can guarantee the failure to happen.

I will put the logic in a method that clearly indicate this is for testing.